server/src/test/java/io/spine/server/BoundedContextTest.java

```diff
@@ -181,15 +195,54 @@ void propagateRepositoriesToStand() {
         assertThat(stand.getExposedTypes(), contains(repository.getEntityStateType()));
     }
 
-    @Test
-    @DisplayName("not allow two aggregate repositories with aggregates of same state")
-    void throwOnSameAggregateState() {
-        ProjectAggregateRepository repository = new ProjectAggregateRepository();
+    @SuppressWarnings("unchecked")
+    @ParameterizedTest
+    @MethodSource("sameStateRepositories")
+    @DisplayName("not allow two entity repositories with entities of same state")
+    void throwOnSameEntityState(Pair<Repository, Repository> sameStateRepositories) {
+        Repository repository = sameStateRepositories.getKey();
         boundedContext.register(repository);
+        Repository anotherRepository = sameStateRepositories.getValue();
+        assertThrows(IllegalStateException.class, () -> boundedContext.register(anotherRepository));
+    }
 
-        AnotherProjectAggregateRepository anotherRepo = new AnotherProjectAggregateRepository();
+    /**
+     * To check whether {@link io.spine.server.BoundedContext} really throws an
+     * {@code IllegalStateException} upon an attempt to register a repository that manages an entity
+     * of a state that a registered entity repository is already managing, all combinations of
+     * entities that take state as a type parameter need to be checked.
+     *
+     * <p>This method returns a stream of pairs of all such combinations, which is a cartesian
```

Comment:
It is worth noting that _all_ the repositories manage entities with the _same_ state type.